visualization/src/pcl_visualizer.cpp

```diff
@@ -3510,16 +3510,20 @@ pcl::visualization::PCLVisualizer::addTextureMesh (const pcl::TextureMesh &mesh,
   vtkTextureUnitManager* tex_manager = vtkOpenGLRenderWindow::SafeDownCast (win_)->GetTextureUnitManager ();
   if (!tex_manager)
     return (false);
-  // Check if hardware support multi texture
+  // hardware always supports multitexturing of some degree
   int texture_units = tex_manager->GetNumberOfTextureUnits ();
-  if ((mesh.tex_materials.size () > 1) && (texture_units > 1))
+  if ((mesh.tex_materials.size () > 1))
   {
     if ((size_t) texture_units < mesh.tex_materials.size ())
       PCL_WARN ("[PCLVisualizer::addTextureMesh] GPU texture units %d < mesh textures %d!\n",
                 texture_units, mesh.tex_materials.size ());
     // Load textures
     std::size_t last_tex_id = std::min (static_cast<int> (mesh.tex_materials.size ()), texture_units);
+#if VTK_MAJOR_VERSION > 5
+    std::string tu = mesh.tex_materials[0].tex_name.c_str ();
```

Comment:
Shouldn't this be `const char*`? Because this the type that `c_str()` returns and also the type that VTK expects? Apart from that looks good.